File: CodeForces And Random Practice Problems/chatroom.py
s = input()
lst = ['h','e','l','l','o']
j = 0
flag = 0
counter = 0
for i in s:
    if(lst[j] == i):
        flag+=1
        j = j+1
    if(j == 5):
        break
        
if(flag == 5):
	print("YES")
else:
	print("NO")

File: CodeForces And Random Practice Problems/chatroom.cpp
#include<bits/stdc++.h>
using namespace std;
int main()
{
    string s;
    cin>>s;
    int counter = 1;
    char lst[] = {'h','e','l','l','o'};
    int j = 0;
    for(int i=0;i<s.length();i++){
        if(lst[j] == s[i])
            j = j+1;
            counter++;}

    cout<<counter;
    if(counter == 4)
        cout<<"YES";
    else
        cout<<"NO";

}
